src/systemTest/java/soot/defaultInterfaceMethods/DefaultInterfaceTest.java

```diff
@@ -0,0 +1,897 @@
+import soot.Body;
+import soot.FastHierarchy;
+import soot.MethodOrMethodContext;
+import soot.PhaseOptions;
+import soot.Scene;
+import soot.SootClass;
+import soot.SootMethod;
+import soot.Unit;
+import soot.UnitPatchingChain;
+import soot.jimple.Stmt;
+import soot.jimple.toolkits.callgraph.Edge;
+import soot.jimple.toolkits.callgraph.ReachableMethods;
+import soot.jimple.toolkits.callgraph.VirtualCalls;
+import soot.testing.framework.AbstractTestingFramework;
+
+/** */
+public class DefaultInterfaceTest extends AbstractTestingFramework {
+
+  @Override
+  protected void setupSoot() {
+    super.setupSoot();
+    PhaseOptions.v().setPhaseOption("cg.cha", "on");
+  }
+
+  @Test
+  public void simpleDefaultInterfaceTest() {
+
+    String testClass = "soot.defaultInterfaceMethods.SimpleDefaultInterface";
+    String defaultClass = "soot.defaultInterfaceMethods.Default";
+
+    final SootMethod target =
+        prepareTarget(
+            methodSigFromComponents(testClass, "void", "main"),
+            testClass,
+            "soot.defaultInterfaceMethods.Default");
+
+    SootMethod defaultMethod =
+        Scene.v().getMethod("<soot.defaultInterfaceMethods.Default: void target()>");
+    Body body = target.retrieveActiveBody();
+    SootMethod targetMethod = resolveMethodRefInBody(body.getUnits(), "void target()");
+    SootMethod resolvedMethod =
+        VirtualCalls.v()
+            .resolveNonSpecial(Scene.v().getRefType(testClass), defaultMethod.makeRef(), false);
+    SootMethod concreteImpl =
+        Scene.v()
+            .getFastHierarchy()
+            .resolveConcreteDispatch(Scene.v().getSootClass(testClass), defaultMethod);
+    Set<SootMethod> abstractImpl =
+        Scene.v()
+            .getFastHierarchy()
+            .resolveAbstractDispatch(Scene.v().getSootClass(defaultClass), defaultMethod);
+
+    boolean edgePresent = checkInEdges(defaultMethod, target);
+    final ReachableMethods reachableMethods = Scene.v().getReachableMethods();
+    /* Arguments for assert function */
+
+    assertEquals(defaultMethod, resolvedMethod);
+    assertEquals(defaultMethod, targetMethod);
+    assertEquals(defaultMethod.getName(), "target");
+    assertNotNull(defaultMethod);
+    assertTrue(reachableMethods.contains(defaultMethod));
+    assertTrue(edgePresent);
+    assertEquals(defaultMethod, concreteImpl);
+    assertTrue(
+        abstractImpl.contains(
+            Scene.v().getMethod("<soot.defaultInterfaceMethods.Default: void target()>")));
+  }
+
+  @Test
+  public void interfaceSameSignatureTest() {
+    String testClassSig = "soot.defaultInterfaceMethods.InterfaceSameSignature";
+    String interfaceReadSig = "soot.defaultInterfaceMethods.Read";
+    String interfaceWriteSig = "soot.defaultInterfaceMethods.Write";
+
+    final SootMethod target =
+        prepareTarget(
+            methodSigFromComponents(testClassSig, "void", "main"),
+            testClassSig,
+            "soot.defaultInterfaceMethods.Read",
```

Comment:
Replace by variables

Reply:
Should I replace for all the test methods?

Reply:
If there are already varaibles existing for these strings then yes! We do not want to have duplicate to be able to more easily change things later.